pennylane/gradients/finite_difference.py

```diff
@@ -347,6 +348,14 @@ def finite_diff(
         shapes.append(len(g_tapes))
 
     def processing_fn(results):
+        # Apply the same squeezing as in qml.QNode to make the transform output consistent.
+        # pylint: disable=protected-access
+        if tape._qfunc_output is not None and not isinstance(tape._qfunc_output, Sequence):
+            squeezed_results = []
+            for res in results:
+                squeezed_results.append(qml.math.squeeze(res))
+            results = squeezed_results

```

Comment:
maybe for brevity, one could do
```suggestion
            results = [qml.math.squeeze(res) for res in results]
```